crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -4,613 +4,609 @@
+public fun supply<T>(treasury: &mut TreasuryCap<T>): &Supply<T> {
+    &treasury.total_supply
+}
 
-    /// This creates a new currency, via `create_currency`, but with an extra capability that
-    /// allows for specific addresses to have their coins frozen. Those addresses cannot interact
-    /// with the coin as input objects.
-    #[deprecated(note = b"For new coins, use `create_regulated_currency_v2`. To migrate existing regulated currencies, migrate with `migrate_regulated_currency_to_v2`")]
-    public fun create_regulated_currency<T: drop>(
-        witness: T,
-        decimals: u8,
-        symbol: vector<u8>,
-        name: vector<u8>,
-        description: vector<u8>,
-        icon_url: Option<Url>,
-        ctx: &mut TxContext
-    ): (TreasuryCap<T>, DenyCap<T>, CoinMetadata<T>) {
-        let (treasury_cap, metadata) = create_currency(
-            witness,
-            decimals,
-            symbol,
-            name,
-            description,
-            icon_url,
-            ctx
-        );
-        let deny_cap = DenyCap {
-            id: object::new(ctx),
-        };
-        transfer::freeze_object(RegulatedCoinMetadata<T> {
-            id: object::new(ctx),
-            coin_metadata_object: object::id(&metadata),
-            deny_cap_object: object::id(&deny_cap),
-        });
-        (treasury_cap, deny_cap, metadata)
-    }
+// deprecated as we have CoinMetadata now
+#[allow(unused_field)]
+public struct CurrencyCreated<phantom T> has copy, drop {
+    decimals: u8,
+}
 
+/// Capability allowing the bearer to freeze addresses, preventing those addresses from
+/// interacting with the coin as an input to a transaction.
+public struct DenyCap<phantom T> has key, store {
+    id: UID,
+}
 
-    /// The index into the deny list vector for the `sui::coin::Coin` type.
-    const DENY_LIST_COIN_INDEX: u64 = 0; // TODO public(package) const
-
-    /// Adds the given address to the deny list, preventing it
-    /// from interacting with the specified coin type as an input to a transaction.
-    #[deprecated(note = b"Use `migrate_regulated_currency_to_v2` to migrate to v2 and then use `deny_list_v2_add`")]
-    public fun deny_list_add<T>(
-       deny_list: &mut DenyList,
-       _deny_cap: &mut DenyCap<T>,
-       addr: address,
-       _ctx: &mut TxContext
-    ) {
-        let `type` =
-            type_name::into_string(type_name::get_with_original_ids<T>()).into_bytes();
-        deny_list.v1_add(
-            DENY_LIST_COIN_INDEX,
-            `type`,
-            addr,
-        )
-    }
+/// This creates a new currency, via `create_currency`, but with an extra capability that
+/// allows for specific addresses to have their coins frozen. Those addresses cannot interact
+/// with the coin as input objects.
+#[
+    deprecated(
+        note = b"For new coins, use `create_regulated_currency_v2`. To migrate existing regulated currencies, migrate with `migrate_regulated_currency_to_v2`",
```

Comment:
Can we split this string over multiple lines? Do we even support such a thing/

Reply:
We don't iirc. I would be happy if we supported multiline strings. Much needed